htdocs/comm/propal/card.php

```diff
@@ -978,36 +978,56 @@
 			}
 			$result = $object->updateline($line->id, $line->subprice, $line->qty, $remise_percent, $tvatx, $line->localtax1_tx, $line->localtax2_tx, $line->desc, 'HT', $line->info_bits, $line->special_code, $line->fk_parent_line, 0, $line->fk_fournprice, $line->pa_ht, $line->label, $line->product_type, $line->date_start, $line->date_end, $line->array_options, $line->fk_unit, $line->multicurrency_subprice);
 		}
-	} elseif ($action == 'addline' && GETPOST('submitforallmargins', 'alpha') && GETPOST('marginforalllines') !== '' && $usercancreate) {
+	} elseif ($action == 'addline' && (GETPOST('submitforallmargins', 'alpha')
```

Comment:
The && and the || are on same level
a && b || cc
You must use parenthesis to be clear in which test you expect.
Also i recommand to move the $usercancreate that is twice just after the
`if ($action == 'addline' && $usercancreate && (...  ||  ...))
`